File: MODUL 6/SOAL 4/PRAK604-2410817210026-FadhilSyahdamaMahatmaPutra.c
#include <stdio.h>
#include <string.h>

void bandingkanArray(char bocchi[], char rover[], int n) {
    int hashCount = 0;  
    int starCount = 0;  
    
    for (int i = 0; i < n; i++) {
        if (bocchi[i] == ' ' && rover[i] == ' ') {
            printf(" ");
        } else if (bocchi[i] == rover[i]) {
            printf("*");
            starCount++;  
        } else {
            printf("#");
            hashCount++;  
        }
    }
    printf("\n");
    printf("* = %d\n", starCount);
    printf("# = %d\n", hashCount);

    if (hashCount > starCount) {
        printf("Pesan Palsu\n");
    } else if (starCount > hashCount) {
        printf("Pesan Asli\n");
    } 
}

int main() {
    char bocchi[100], rover[100];
    int n1, n2;

    fgets(bocchi, sizeof(bocchi), stdin);
    fgets(rover, sizeof(rover), stdin);

    bocchi[strcspn(bocchi, "\n")] = '\0';
    rover[strcspn(rover, "\n")] = '\0';

    n1 = strlen(bocchi);
    n2 = strlen(rover);

    if (n1 != n2) {
        printf("Panjang kalimat berbeda, pesan palsu\n");
    } else {
        bandingkanArray(bocchi, rover, n1);
    }
}

File: MODUL 6/SOAL 4/PRAK604-2410817210026-FadhilSyahdamaMahatmaPutra.py
def bandingkan_array(bocchi, rover):
    hash_count = 0
    star_count = 0
    
    for i in range(len(bocchi)):
        if bocchi[i] == ' ' and rover[i] == ' ':
            print(" ", end="")
        elif bocchi[i] == rover[i]:
            print("*", end="")
            star_count += 1
        else:
            print("#", end="")
            hash_count += 1

    print()
    print(f"* = {star_count}")
    print(f"# = {hash_count}")

    if hash_count > star_count:
        print("Pesan Palsu")
    elif star_count > hash_count:
        print("Pesan Asli")

bocchi = input().rstrip()
rover = input().rstrip()

if len(bocchi) != len(rover):
    print("Panjang kalimat berbeda, pesan palsu")
else:
    bandingkan_array(bocchi, rover)